pkg/catalog/gateway.go

```diff
@@ -0,0 +1,14 @@
+package catalog
+
+import (
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/identity"
+)
+
+// IsMultiClusterGateway checks if the ServiceIdentity belongs to the MultiClusterGateway.
+// Only used if MultiClusterMode is enabled.
+func (mc *MeshCatalog) IsMultiClusterGateway(svcIdentity identity.ServiceIdentity) bool {
+	sa := svcIdentity.ToK8sServiceAccount()
+	return mc.configurator.GetFeatureFlags().EnableMulticlusterMode &&
+		envoy.ProxyKind(sa.Name) == envoy.KindGateway && sa.Namespace == mc.configurator.GetOSMNamespace()
```

Comment:
Any reason why the namespace check is necessary here, I don't see how a gateway can connect over TLS to the osm-controller and not be a part of the correct namespace.

Reply:
It prevents a user's pod/sidecar with a serviceAccount named gateway (in a separate, but monitored namespace) from being treated as a gateway.

Reply:
I see the problem here, you aren't leveraging the proxy kind from the proxy metadata, but rather from it's workload cert which doesn't have the Kind encoded. 
This is okay for now, but the caller (some XDS code) is already aware this is a gateway, so this check can be completely but the current implementation in this PR would need to change.